ft_strtrim.c

```diff
@@ -0,0 +1,101 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   ft_strtrim.c                                       :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: ldamiens <ldamiens@student.42.fr>          +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2021/11/07 15:08:03 by ldamiens          #+#    #+#             */
+/*   Updated: 2021/11/11 19:50:26 by ldamiens         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "libft.h"
+#include <stdlib.h>
+
+int	contains(char const *array, char character)
+{
+	int	index;
+
+	index = 0;
+	while (array[index])
+	{
+		if (array[index] == character)
+			return (1);
+		index++;
+	}
+	return (0);
+}
+
+void	reverse(char *string)
+{
+	int		index;
+	int		len;
+	char	tmp;
+
+	index = 0;
+	len = ft_strlen(string);
+	while (index < len / 2)
+	{
+		tmp = string[index];
+		string[index] = string[len - index - 1];
+		string[len - index - 1] = tmp;
+		index++;
+	}
+}
+
+int	getstart(char *s1, char const *set)
+{
+	int	index;
+
+	index = 0;
+	while (s1[index])
+	{
+		if (!contains(set, s1[index]))
+			return (index);
+		index++;
+	}
+	return (0);
+}
+
+int	getend(char *s1, char const *set)
+{
+	int	index;
+
+	index = 0;
+	reverse(s1);
+	while (s1[index])
+	{
+		if (!contains(set, s1[index]))
+		{
+			reverse(s1);
+			return (ft_strlen(s1) - index - 1);
+		}
+		index++;
+	}
+	reverse(s1);
+	return (ft_strlen(s1) - 1);
+}
```

Comment:
Pretty bad implementation
```suggestion
static size_t getend(char *s1, char const *set)
{
	size_t index;
	
	index = ft_strlen(s1);
	while (index--)
		if (!contains(set, s1[index])
			break;
	return index;
}
```